app/services/katello/smart_proxy_helper.rb

```diff
@@ -13,34 +13,32 @@ def default_capsule?
       @smart_proxy.pulp_master?
     end
 
-    def repos_available_to_capsule(environment = nil, content_view = nil, repository = nil)
-      ret = []
-      if repository
-        environment = repository.environment
-        ret = [repository]
-      else
-        yum_repos = fetch_repos_available_to_capsule(environment, content_view) || []
-        puppet_envs = fetch_puppet_environments_available_to_capsule(environment, content_view) || []
-        ret = yum_repos + puppet_envs
-      end
+    def lifecycle_environment_check(environment = nil, repository = nil)
+      environment = repository.environment if repository
 
       if environment && !self.smart_proxy.lifecycle_environments.include?(environment)
         fail _("Lifecycle environment '%{environment}' is not attached to this capsule.") % { :environment => environment.name }
       end
-
-      ret
     end
 
-    private
+    def combined_repos_available_to_capsule(environment = nil, content_view = nil, repository = nil)
+      lifecycle_environment_check(environment, repository
+      )
```

Comment:
Was having this closing ) on a separate line intentional?

Reply:
nope!